src/Xamarin.Android.Build.Tasks/MSBuild/Xamarin/Android/Microsoft.Android.Sdk.Bindings.Gradle.targets

```diff
@@ -0,0 +1,125 @@
+        Condition=" '@(AndroidGradleProject->Count())' != '0' ">
+    <ItemGroup>
+      <_AllGradleFiles Include="%(AndroidGradleProject.RootDir)%(AndroidGradleProject.Directory)/**/*"
+          Condition= " Exists('%(AndroidGradleProject.RootDir)%(AndroidGradleProject.Directory)') "/>
+      <_GradleInputs Include="@(_AllGradleFiles)"
+          Condition="  '%(Extension)' == '.java'
+                    or '%(Extension)' == '.kt'
+                    or '%(Extension)' == '.kts'
+                    or '%(Extension)' == '.gradle'
+                    or '%(Extension)' == '.properties'
+                    or '%(Extension)' == '.xml' " />
+      <_GradleInputs Include="$(MSBuildProjectFullPath)" />
+      <_GradleInputs Include="@(_AndroidMSBuildAllProjects)" />
+    </ItemGroup>
+  </Target>
+
+  <Target Name="_CalculateAndroidGradleProjectOutputPath"
+        Condition=" '@(AndroidGradleProject->Count())' != '0' "
+        Outputs="%(AndroidGradleProject.Identity)" >
+    <Hash
+        ItemsToHash="%(AndroidGradleProject.Identity)"
+        IgnoreCase="true">
+      <Output TaskParameter="HashResult" PropertyName="_AndroidGradleProjectHash" />
+    </Hash>
+    <ItemGroup>
+      <AndroidGradleProject Condition=" '%(AndroidGradleProject.OutputPath)' == '' " >
+        <OutputPath>$(_AndroidGradleDefaultOutputPathRoot)%(ModuleName)-$([System.String]::Copy($(_AndroidGradleProjectHash)).Substring(0, 5))/</OutputPath>
+      </AndroidGradleProject>
+    </ItemGroup>
+  </Target>
+
+  <Target Name="_BuildAndroidGradleProjects"
+      Condition=" '@(AndroidGradleProject->Count())' != '0' "
+      DependsOnTargets="_SetupMSBuildAllProjects;AndroidPrepareForBuild;_GetBuildAndroidGradleProjectsInputs;_CalculateAndroidGradleProjectOutputPath"
+      Inputs="@(_GradleInputs)"
+      Outputs="$(_BuildAndroidGradleProjectsStamp)" >
+    <!-- Delete outputs folder if inputs have changed to force a partial rebuild -->
+    <RemoveDir Directories="@(AndroidGradleProject->'%(OutputPath)outputs')" />
+
+    <!-- Create the net.android.init.gradle.kts script used to override the Gradle project output directory -->
+    <CopyResource
+        ResourceName="net.android.init.gradle.kts"
+        OutputPath="$(_AndroidGradleBuildDirInitScriptPath)"
+    />
+
+    <!-- Run assemble task for project outputs, android app and library project outputs are currently supported -->
+    <Gradle ToolPath="%(AndroidGradleProject.RootDir)%(AndroidGradleProject.Directory)"
+        BuildDirInitScriptPath="$(_AndroidGradleBuildDirInitScriptPath)"
+        Command="assemble%(AndroidGradleProject.Configuration)"
+        ModuleName="%(AndroidGradleProject.ModuleName)"
+        OutputPath="%(AndroidGradleProject.OutputPath)"
+        AndroidSdkDirectory="$(AndroidSdkDirectory)"
+        JavaSdkDirectory="$(JavaSdkDirectory)" >
+    </Gradle>
+
+    <ItemGroup>
+      <_AndroidGradleProjectAppOutputs      Include="%(AndroidGradleProject.OutputPath)outputs/**/*.apk" />
```

Comment:
This maywell pick up both Relase and Debug outputs if I remember gradle correctly. Is that what we want? We might get a debug and releae verison of the same apk or aar.

Reply:
@dellis1972: The only way I can think of for this to happen is if:

 1. The user sets `%(OutputPath)` explicitly, and
 2. The user has done manual debug and release builds into the path set by (1).

Which is certainly something that can happen, and maybe part of [why I wondered about whether or not `%(OutputPath)` should be documented at all](https://github.com/dotnet/android/pull/9270#discussion_r1765391732), but this doesn't seem like it'll be particularly *common* either.

Reply:
Looking at this again, the Gradle project configuration can be set via the `%(Configuration)` metadata as well. The way the targets are set up the output directory should be cleared when this value is changed or set, but it may still be worth adding `%(AndroidGradleProject .Configuration)` to the default gradle project output path